adapters/adorsys-integ-adapter/src/test/java/de/adorsys/xs2a/adapter/adorsys/service/PaymentInitiationServiceWireMockTest.java

```diff
@@ -0,0 +1,280 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.adorsys.xs2a.adapter.adorsys.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import de.adorsys.xs2a.adapter.adapter.BasePaymentInitiationService;
+import de.adorsys.xs2a.adapter.adapter.link.identity.IdentityLinksRewriter;
+import de.adorsys.xs2a.adapter.api.model.*;
+import de.adorsys.xs2a.adapter.http.ApacheHttpClient;
+import de.adorsys.xs2a.adapter.http.HttpClient;
+import de.adorsys.xs2a.adapter.http.JsonMapper;
+import de.adorsys.xs2a.adapter.service.PaymentInitiationService;
+import de.adorsys.xs2a.adapter.service.RequestHeaders;
+import de.adorsys.xs2a.adapter.service.RequestParams;
+import de.adorsys.xs2a.adapter.service.Response;
+import de.adorsys.xs2a.adapter.service.link.LinksRewriter;
+import de.adorsys.xs2a.adapter.service.model.Aspsp;
+import org.apache.http.HttpStatus;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import pro.javatar.commons.reader.JsonReader;
+import pro.javatar.commons.reader.ResourceReader;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class PaymentInitiationServiceWireMockTest {
+
+    public static final String PAYMENT_ID = "W_pFk5-4OqzsXpxKLs9h97q8bfPnV3XKAm5MxM8dnT60LxXyPaGedv4HPQtEQ8-mcgftJbETkzvNvu5mZQqWcA==_=_psGLvQpt9Q";
+    public static final String AUTHORISATION_ID = "259b8215-d14e-493a-ba01-c2243a9ff86a";
+    public static final String PERIODIC_PAYMENT_ID = "RWPbX-Qgnmjb5yixTXwywIU9xxlutmt70MthNORZI9pVXXwjlE_8wK5HaZwohPtAcgftJbETkzvNvu5mZQqWcA==_=_psGLvQpt9Q";
+    public static final String PERIODIC_AUTHORISATION_ID = "946d8445-7548-43e0-8cfc-f092d7ebc6cb";
+
+    private final ObjectMapper objectMapper = new JsonMapper().getNewInstance();
+    private ResourceReader reader = JsonReader.getInstance(objectMapper);
+
+    private static PaymentInitiationService service;
+
+    //No-args constructor will start on port 8080, no HTTPS
+    private static WireMockServer wireMockServer;
+
+    @BeforeAll
+    static void beforeAll() {
+        wireMockServer = new WireMockServer(wireMockConfig().port(8189));
+        wireMockServer.start();
+
+
+        HttpClient httpClient = new ApacheHttpClient(HttpClientBuilder.create().build());
+        LinksRewriter linksRewriter = new IdentityLinksRewriter();
+        Aspsp aspsp = new Aspsp();
+        aspsp.setUrl("http://localhost:" + wireMockServer.port());
+
+        service = new BasePaymentInitiationService(aspsp, httpClient, null, linksRewriter);
```

Comment:
perhaps the service should be retrieved from the service provider

Reply:
replaced, thanks